### ISSUE TITLE
[Issue] Partial upgrade in the install script

### ISSUE THREAD

**Issue by cd-CreepArghhh** [end-4/dots-hyprland#363]
Note: I have ignored the template as it is not relevant for this issue.

### The issue

[/scriptdata/installers, L11](https://github.com/end-4/dots-hyprland/blob/main/scriptdata/installers#L11):

```
x sudo pacman -Sy --needed --noconfirm base-devel
```

I'm filing an issue because this *is* an issue: if yay isn't installed and base-devel isn't already up to date, depending on when this script is run, if base-devel isn't already up to date, there could be problems. There's a reason why [the Arch wiki tells you not to run partial upgrades](https://wiki.archlinux.org/title/System_maintenance#Partial_upgrades_are_unsupported).

### Recommended fix

`sudo pacman -Syu` before starting

Of course, if you have a reason to run `-Sy` (maybe it upgrades the system somewhere else and I somehow missed the line), feel free to close this issue as wontfix.
